### PR TITLE
Add location service of postnl

### PR DESCRIPTION
**Pull request by Jomaanro**
I've added the basic functionality for the location service provided by PostNL. It will locate the nearest dropoff point from a zipcode. In the future I will add the remaining two functions (locations in area and location info)

### CODE REVIEW

File: src/Postnl.php [hunk 1]
```diff
@@ -315,6 +315,20 @@ public function currentStatus($barcode)
...
 
     /**
+     * @param $PostcalCode
+     * @return mixed
+     * @throws ComplexTypes\CifException
+     * @throws SoapFault
+     */
+    public function GetNearestLocation($PostcalCode, $AllowSundaySorting = "false", $DeliveryDate = null) {
+        $message = new ComplexTypes\Message;
+        $location = new ComplexTypes\Location($PostcalCode, $AllowSundaySorting, $DeliveryDate);
```
Comment (slokhorst) [inline src/Postnl.php]: Typo: `PostcalCode` (also 2 lines below)

**Comment by slokhorst** [comment]
Looks really useful!

A small style note: the project uses lower case (first letter) variables everywhere, but this pull uses upper case. Would be nice to be consistent.

**Comment by Jomaanro** [comment]
Thanks for reviewing. Updated the functions to use correct camelCasing. I see that variables do have uppercase in all the other files so I left it like that. Also updated some more comment sections.

**Comment by slokhorst** [comment]
Ah sorry you're right, I only looked at `Postnl.php`. Looks good!

**Comment by ameenross** [comment]
Yeah, I kinda hated the camelcased variables. I think I had to do it because of how SoapClient serializes an object to XML.

**Comment by reijervano** [comment]
Nice work! Will you accept the request soon?

File: src/ComplexTypes/GetNearestLocationsRequest.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+     * @var string $Countrycode
+     */
+    protected $Countrycode = 'NL';
+
+    /**
+     * GetNearestLocationsRequest constructor.
+     * @param Message $Message
+     * @param Location $Location
+     */
+    public function __construct(
```
Comment (ameenross) [inline src/ComplexTypes/GetNearestLocationsRequest.php:26]: The parameters should be a single line

File: src/ComplexTypes/GetNearestLocationsRequest.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    {
+        return $this->Message;
+    }
+
+    /**
+     * @param Message $Message
+     */
+    public function setMessage($Message)
+    {
+        $this->Message = $Message;
```
Comment (ameenross) [inline src/ComplexTypes/GetNearestLocationsRequest.php:47]:
``` php
return $this;
```

For chaining methods

File: src/ComplexTypes/Location.php [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+    /**
+     * @var string $Postalcode
+     */
+    protected $Postalcode = null;
+
+    /**
+     * @var array $DeliveryOptions
+     */
+    protected $DeliveryOptions = array();
```
Comment (ameenross) [inline src/ComplexTypes/Location.php:29]: Use short array syntax. I require 5.4 anyway

File: src/ComplexTypes/Location.php [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    /**
+     * Location constructor.
+     * @param $Postalcode
+     * @param string $AllowSundaySorting
+     * @param date $DeliveryDate
+     */
+    public function __construct($Postalcode, $AllowSundaySorting = "false", $DeliveryDate = null)
+    {
+        $this->setAllowSundaySorting($AllowSundaySorting);
+        $this->setDeliveryDate($DeliveryDate ?: date('d-m-Y',strtotime('next monday')));
```
Comment (ameenross) [inline src/ComplexTypes/Location.php:46]: A space after comma.

Also, I'd prefer OOP method:

``` php
(new DateTime('next monday'))->format('d-m-Y')
```

File: src/LocationClient.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+     */
+    protected $classes = [
+        'GetNearestLocation',
+    ];
+
+    public function getNearestLocation(ComplexTypes\GetNearestLocationsRequest $getNearestLocations)
+    {
+        return $this->__soapCall('GetNearestLocations', [$getNearestLocations]);
+    }
+}
```
Comment (ameenross) [inline src/LocationClient.php:31]: Newline at end of file

File: src/LocationClient.php [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+     * The URL of the sandbox WSDL.
+     */
+    const SANDBOX_WSDL = 'https://testservice.postnl.com/CIF_SB/LocationWebService/2_0/?wsdl';
+
+    /**
+     * @var array $classes
+     *     The complex types used by this client.
+     */
+    protected $classes = [
+        'GetNearestLocation',
```
Comment (ameenross) [inline src/LocationClient.php:24]: I think there should be a lot more, like `Message`.

File: src/Postnl.php [hunk 8]
```diff
@@ -315,6 +315,36 @@ public function currentStatus($barcode)
...
 
     /**
+     * @param $postalCode
+     * @param string $allowSundaySorting
+     * @param null $deliveryDate
+     * @return mixed
+     * @throws ComplexTypes\CifException
+     * @throws SoapFault
+     */
+    public function getNearestLocation($postalCode, $allowSundaySorting = "false", $deliveryDate = null) {
```
Comment (ameenross) [inline src/Postnl.php:325]: Opening brace on next line

**Comment by danslo** [comment]
Any update on this PR? I'll probably need this functionality relatively soon. I can update the styling issues if you haven't started on it yet.

**Comment by ameenross** [comment]
You're welcome to. You'd have to create a branch off of @Jomaanro's branch though.

**Pull request closed without merging**